M2/Macaulay2/d/parser.d

```diff
@@ -518,7 +525,10 @@ export treePosition(e:ParseTree):Position := (
     is s:Parentheses      do combinePositionL(s.left.position,       s.right.position)
     is s:EmptyParentheses do combinePositionL(s.left.position,       s.right.position)
     is a:Adjacent         do combinePositionM(treePosition(a.lhs),   treePosition(a.rhs))
-    is a:Arrow            do combinePositionL(treePosition(a.lhs),   treePosition(a.rhs))
+    is a:Arrow            do (
```

Comment:
Treating `TEST` as `->` is kludgy.

Reply:
Yeah, it is a little strange.  We need to call `bind` when things are still `ParseTree` objects in order to create a local dictionary, which is why I went this route.  And `Arrow` is already set up for creating functions.

I also considered creating another member of `ParseTree`.  Then we could avoid all the `when lhs is dummy` cases.